pkg/graph/resolvers/system_intake_grb_reviewer.go

```diff
@@ -122,6 +122,42 @@ func DeleteSystemIntakeGRBReviewer(
 	})
 }
 
+func CastSystemIntakeGRBReviewerVote(ctx context.Context, store *storage.Store, input models.CastSystemIntakeGRBReviewerVoteInput) (*models.SystemIntakeGRBReviewer, error) {
+	// first, if "OBJECT" is the vote selection, confirm there is a comment (required for objections)
+	if input.Vote == models.SystemIntakeAsyncGRBVotingOptionObjection && (input.VoteComment == nil || len(*input.VoteComment) < 1) {
+		return nil, errors.New("vote comment is required with an `Objection` vote")
+	}
+
+	// then, check if the GRB review is in a state where votes are allowed - do this second to avoid a db round trip
+	// if the above condition isn't met
+	// get system intake
+	systemIntake, err := store.FetchSystemIntakeByID(ctx, input.SystemIntakeID)
```

Comment:
NIT: could this use an existing data loader instead? Ok to hold off for a follow up PR.